### PR TITLE
More knobs added to limit-pool

### PR DESCRIPTION
**Pull request by danboykis**
It's now possible to choose between Array or Linked queues, as well as rejected-handler.
Pool threads are now named and numbered.
'lasync' namespace has been changed to 'lasync.core' as well as show.clj is now moved out to a dev area.

### CODE REVIEW

File: src/java/lasync/limitq/LinkedLimitedQueue.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package lasync.limitq;
+
+import java.util.concurrent.LinkedBlockingDeque;
+
+public class LinkedLimitedQueue<E> extends LinkedBlockingDeque<E> {
```
Comment (tolitius) [inline src/java/lasync/limitq/LinkedLimitedQueue.java:5]: needs a constructor with capacity

File: project.clj [hunk 2]
```diff
@@ -4,11 +4,15 @@
...
   :java-source-paths ["src/java"]          
 
   :scm {:name "git"
         :url "https://github.com/tolitius/lasync"}
 
+  :profiles {:dev {:source-paths ["dev"]
+                   :repl-options {:init-ns show}}}
+
   :dependencies [[org.clojure/tools.logging "0.2.6"]
+                 [org.clojure/tools.logging "0.3.1"]
```
Comment (tolitius) [inline project.clj]: no need for 2 tools.logging

File: README.md [hunk 3]
```diff
@@ -102,7 +102,7 @@ the queue max size always stays at 4, and lasync creates that back pressure in c
 In fact the "blocking" can be seen in action, as each task is sleeping for a second, 
 so the whole thing can be visually seen being processed by 4, pause, next 4, pause, etc..
 
-Here is [the code](https://github.com/tolitius/lasync/blob/master/src/lasync/show.clj) behind the show
+Here is [the code](dev/show.clj) behind the show
```
Comment (tolitius) [inline README.md:105]: nice going :)

File: src/lasync/core.clj [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+        ueh (thread-exception-handler)]
+    (reify ThreadFactory
+      (newThread [_ runnable]
+        (let [t (Thread. runnable)]
+          (doto t
+            (.setName (str name "-" (.incrementAndGet counter)))
+            (.setDaemon true)
+            (.setUncaughtExceptionHandler ueh)))))))
+
+(defn limit-pool [& {:keys [nthreads queue thread-factory rejected-handler]
```
Comment (tolitius) [inline src/lasync/core.clj:36]: api: how do we choose between the queue implementation?

Comment (tolitius) [inline src/lasync/core.clj:36]: ah.. `:queue`, k :)

Comment (tolitius) [inline src/lasync/core.clj:36]: why `ArrayLimitedQueue` is the default? what is the thinking behind it?

Comment (danboykis) [inline src/lasync/core.clj:36]: ArrayLimitedQueue is faster than LinkedLimitedQueue. Much better locality.

**Comment by tolitius** [comment]
would also be good to get a pull request with updated docs :)

**Pull request merged**